plugins/telemetry/telemetry.h

```diff
@@ -6,48 +6,90 @@ namespace dronecore {
 
 class TelemetryImpl;
 
+/**
+ * The Telemetry class allows to get all kinds of telemetry infos from a device.
+ */
 class Telemetry
 {
 public:
+    /**
+     * Constructor for Telemetry called internally.
+     */
     explicit Telemetry(TelemetryImpl *impl);
+
+    /**
+     * Destructor for Telemetry called internally.
+     */
     ~Telemetry();
 
+    /**
+     * Position type.
+     */
     struct Position {
-        double latitude_deg;
-        double longitude_deg;
-        float absolute_altitude_m;
-        float relative_altitude_m;
+        double latitude_deg; /**< latitude from -90 to +90 in degrees */
+        double longitude_deg; /**< longitude from -180 to 180 in degrees */
+        float absolute_altitude_m; /**< altitude AMSL (above mean sea level in meters */
+        float relative_altitude_m; /**< altitude relative to takeoff altitude in meters */
     };
 
+    /**
+     * Quaternion type
+     *
+     * All rotations and axis systems follow the right-hand rule.
+     * The Hamilton quaternion product definition is used.
+     * A zero-rotation quaternion is represented by (1,0,0,0).
+     * The quaternion could also be written as w + xi + yj + zk.
+     */
     struct Quaternion {
-        float w;
-        float x;
-        float y;
-        float z;
+        float w; /**< quaternion entry 0 also denoted as a */
+        float x; /**< quaternion entry 1 also denoted as b */
+        float y; /**< quaternion entry 2 also denoted as c */
+        float z; /**< quaternion entry 3 also denoted as d */
     };
 
+    /**
+     * Euler angle type
```

Comment:
This means nothing to me. Can we have a link to more information? (for readers)

Reply:
Same.